CMakeLists.txt

```diff
@@ -55,59 +55,60 @@ target_link_libraries (nanocat nanomsg)
 
 set (all_tests "")
 
-macro (add_libnanomsg_test NAME)
+macro (add_libnanomsg_test NAME TIMEOUT)
     list (APPEND all_tests ${NAME})
     add_executable (${NAME} tests/${NAME}.c)
     target_link_libraries (${NAME} nanomsg)
     add_test (NAME ${NAME} COMMAND ${NAME})
-    set_tests_properties (${NAME} PROPERTIES  TIMEOUT 30)
+    set_tests_properties (${NAME} PROPERTIES TIMEOUT ${TIMEOUT})
 endmacro (add_libnanomsg_test)
 
 #  Transport tests.
-add_libnanomsg_test (inproc)
-add_libnanomsg_test (inproc_shutdown)
-add_libnanomsg_test (ipc)
-add_libnanomsg_test (ipc_shutdown)
-add_libnanomsg_test (ipc_stress)
-add_libnanomsg_test (tcp)
-add_libnanomsg_test (tcp_shutdown)
-add_libnanomsg_test (ws)
-add_libnanomsg_test (tcpmux)
+add_libnanomsg_test (inproc 5)
+add_libnanomsg_test (inproc_shutdown 5)
+add_libnanomsg_test (ipc 5)
+add_libnanomsg_test (ipc_shutdown 30)
+add_libnanomsg_test (ipc_stress 5)
+add_libnanomsg_test (tcp 5)
+add_libnanomsg_test (tcp_shutdown 120)
```

Comment:
That correlation note is very interesting indeed.  Maybe we are spinning or the test needs to be changed to be less sensitive to load?